xapian-letor/api/feature_internal.cc

```diff
@@ -0,0 +1,181 @@
+/** @file feature_internal.cc
+ * @brief Definition of Feature::Internal class.
+ */
+/* Copyright (C) 2019 Vaibhav Kansagara
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+
+#include <config.h>
+
+#include "xapian-letor/feature.h"
+#include "feature_internal.h"
+
+#include <cmath>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include "debuglog.h"
+
+using namespace std;
+using namespace Xapian;
+
+Database
+Feature::Internal::get_database() const
+{
+    LOGCALL(API, Database, "Feature::Internal::get_database", NO_ARGS);
+    return feature_db;
+}
+
+Query
+Feature::Internal::get_query() const
+{
+    LOGCALL(API, Query, "Feature::Internal::get_query", NO_ARGS);
+    return feature_query;
+}
+
+Document
+Feature::Internal::get_doc() const
+{
+    LOGCALL(API, Document, "Feature::Internal::get_doc", NO_ARGS);
+    return feature_doc;
+}
+
+bool
+Feature::Internal::get_termfreq(const std::string & term,
+			double & termfreq_) const
```

Comment:
The wrapped line isn't indented enough.

And why is this returning `double`?  `termfreq` should be type `Xapian::doccount`.

Also preferred Xapian style for writing the `&` in reference types (and the `*` in pointer types) is what seems to be the standard C++ style of putting no space before but a space after, i.e. `const std::string& term` and `double& termfreq_` here.  Quite a lot of older code doesn't conform to this - fixing everything by hand is a lot of work, and we've failed to find an automatic code formatting tool which does a good enough job.  But I'd really like any new code to follow this preferred style.

Reply:
fixed in a fixup commit.